src/main/host/descriptor/epoll/mod.rs

```diff
@@ -371,6 +371,9 @@ impl Epoll {
         // Add everything that is still ready back to the ready set.
         self.ready.extend(keep);
 
+        // We've mutated the ready list; we may need to trigger callbacks.
+        CallbackQueue::queue_and_run(|cb_queue| self.refresh_state(cb_queue));
```

Comment:
I think the `CallbackQueue` should be passed in as an argument so that the closure will start before the `borrow_mut()` call. So `epoll_wait_helper` would have a:

```rust
let ready = CallbackQueue::queue_and_run(|cb_queue| epoll.borrow_mut().collect_ready_events(max_events, cb_queue));
```